JAICore/jaicore-ml/src/jaicore/ml/cache/ReproducibleInstances.java

```diff
@@ -0,0 +1,175 @@
+package jaicore.ml.cache;
+
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+
+import jaicore.ml.WekaUtil;
+import jaicore.ml.openml.OpenMLHelper;
+import weka.core.Instances;
+
+/**
+ * New Instances class to track splits and data origin.
```

Comment:
Please add more context here. Imagine you have no idea what the class is doing - would you understand its purpose with this javadoc?